ui/src/app/pages/data/data-set/genomic-extraction-modal.spec.tsx

```diff
@@ -0,0 +1,117 @@
+  let dataset;
+  let datasetApiStub;
+  let testProps;
+  let workspace;
+
+  const component = () => {
+    return <GenomicExtractionModal {...testProps}/>;
+  }
+
+  beforeEach(() => {
+    datasetApiStub = new DataSetApiStub();
+    registerApiClient(DataSetApi, datasetApiStub);
+
+    dataset = datasetApiStub.getDatasetMock;
+    workspace = workspaceDataStub;
+
+    testProps = {
+      dataSet: dataset,
+      workspaceNamespace: workspace.namespace,
+      workspaceFirecloudName: workspace.id,
+      closeFunction: () => {},
+      title: 'Top 10 Egregious Hacks Your Tech Lead Doesn\'t Want You To Know'
+    }
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should render', () => {
+    const wrapper = mount(component());
+    expect(wrapper.exists()).toBeTruthy();
+  });
+
+  it('should show a warning when there is a currently running extract for this dataset', () => {
+    const oneHourAgo = new Date();
+    oneHourAgo.setHours(oneHourAgo.getHours() - 1);
+    genomicExtractionStore.set({
+      [workspaceDataStub.namespace]: [
+        {
+          status: TerraJobStatus.RUNNING,
+          datasetName: dataset.name,
+        },
+        {
+          status: TerraJobStatus.SUCCEEDED,
+          completionTime: oneHourAgo.getTime(),
+          datasetName: dataset.name,
+        },
+        {
+          status: TerraJobStatus.FAILED,
+          completionTime: oneHourAgo.getTime(),
+          datasetName: dataset.name,
+        }
+      ]
+    });
+
+    const wrapper = mount(component());
+    expect(wrapper.find('[data-test-id="running-extract-warning"]')).toBeTruthy();
+  });
+
+  it('should show a warning message when there is a preexisting completed extract for the dataset', () => {
+    const oneHourAgo = new Date();
+    oneHourAgo.setHours(oneHourAgo.getHours() - 1);
+    genomicExtractionStore.set({
+      [workspaceDataStub.namespace]: [
+        {
+          status: TerraJobStatus.SUCCEEDED,
+          completionTime: oneHourAgo.getTime(),
+          datasetName: dataset.name,
+        },
+        {
+          status: TerraJobStatus.FAILED,
+          completionTime: new Date().getTime(),
+          datasetName: dataset.name,
+        },
+        {
+          status: TerraJobStatus.RUNNING,
+          datasetName: 'some other data set with a different name',
+        }
+      ]
+    });
+
+    const wrapper = mount(component());
+    expect(wrapper.find('[data-test-id="preexisting-extract-warning"]')).toBeTruthy();
+  });
+
+  it('should not show a warning message when there are only failed extracts for this dataset', () => {
+    genomicExtractionStore.set({
+      [workspaceDataStub.namespace]: [
+        {
+          status: TerraJobStatus.FAILED,
+          completionTime: new Date().getTime(),
+          datasetName: dataset.name,
+        },
+        {
+          status: TerraJobStatus.RUNNING,
+          datasetName: 'some other data set with a different name',
```

Comment:
What happens if I swap the names of my datasets? We may need to send the dataset id through the API if there's a bug with name swapping.

Reply:
ooh, good point. you mean if you change the dataset names after the extract is done?

Reply:
ok, we store the dataset id under the hood. so the name will be correct even if the dataset is renamed.

Reply:
ah, right. So even if we rename the datasets the `getExtractionJobs()` call will return the new name. Comparing the name still feels a little iffy to me when we're really checking for same identity but this is fine since there's no bug.

Reply:
correct. and we don't allow datasets with the same names in the same workspace.